### PR TITLE
add design doc for support task Dag with gang

### PR DESCRIPTION
**Pull request by HecarimV**
add design doc for : #2298

### CODE REVIEW

File: docs/design/task-dag-with-gang.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+   - CheckTaskStarving (check whether job has at least one task which is starving)
+2. pg cannot keep `Inqueue` status until task Dag finish.
+3. enqueue resource should not always be pg.minResource
+4. `vcjob.spec.minAvailable`  should be at least greater than `taskMinAvailableTotal`
+
+
+
+## Solution
+
+Because of task dependencies, all task pods will not be created together even if resource requirements are met. However, the check of gang is basically for all tasks under the job. So we can skip the check for unstarted tasks,  that is, to treat started tasks as a whole.
```
Comment (Thor-wl) [inline docs/design/task-dag-with-gang.md:27]: Hey, @HecarimV Thanks for your contribution. I'm sorry for not understanding the solution. Can you give more details about that?

Comment (HecarimV) [reply docs/design/task-dag-with-gang.md:27]: I did not add some new logic, just added some judgments to the gang check, the purpose is to allow the scheduler to process jobs with task dependencies like other normal jobs.

Comment (Thor-wl) [reply docs/design/task-dag-with-gang.md:27]: IC. I think we should take more discussion about this implementation.
@William-wang @shinytang6 Looking forward to see your advice.

**Comment by HecarimV** [comment]
Closed for it doesn't seem so good.

**Pull request closed without merging**